shared_model/backend/protobuf/proto_proposal_factory.hpp

```diff
@@ -20,17 +21,17 @@ namespace shared_model {
       FactoryResult<std::unique_ptr<interface::Proposal>> createProposal(
           interface::types::HeightType height,
           interface::types::TimestampType created_time,
-          const interface::types::TransactionsCollectionType &transactions)
-          override {
+          boost::any_range<interface::Transaction, boost::forward_traversal_tag>
```

Comment:
Why change the type?

Reply:
It is otherwise not possible to apply `boost::adaptors::transformed`:
https://github.com/hyperledger/iroha/blob/bcfb800ac296892624b9fb87f8b7ffcaf744199b/irohad/validation/impl/stateful_validator_impl.cpp#L238-L241
This will be refactored with `using` to avoid a convoluted type in interface.